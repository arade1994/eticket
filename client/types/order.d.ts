import { type Ticket } from "./ticket";

export interface Order {
  id: string;
  userId: string;
  status: OrderStatus;
  expiresAt: Date;
  version: number;
  ticket: Ticket;
}

export declare enum OrderStatus {
  Created = "created",
  Cancelled = "cancelled",
  AwaitingPayment = "awaiting:payment",
  Complete = "complete",
}
