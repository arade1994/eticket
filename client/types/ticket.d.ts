export interface Ticket {
  id: string;
  title: string;
  price: number;
  userId: string;
  category: string;
  version: number;
}
