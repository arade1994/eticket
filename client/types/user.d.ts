import { type IncomingMessage } from "http";

export interface User {
  id: string;
  firstName: string;
  lastName: string;
  age: number;
  email: string;
  password: string;
}

export interface Rating {
  id: string;
  comment: string;
  rate: number;
  userId: string;
  ratedUserId: string;
}

export interface RequestWithUser extends IncomingMessage {
  currentUser?: { id: string; email: string };
}
