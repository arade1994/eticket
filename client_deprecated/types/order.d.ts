import { type Ticket } from "./ticket";

export interface Order {
  id: string;
  userId: string;
  status: OrderStatus;
  expiresAt: string;
  version: number;
  ticket: Ticket;
}

export declare enum OrderStatus {
  Created = "created",
  Cancelled = "cancelled",
  AwaitingPayment = "awaiting:payment",
  Complete = "complete",
}
