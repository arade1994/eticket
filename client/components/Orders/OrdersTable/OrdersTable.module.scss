.ordersTable {
  width: 100%;
  border: 1px solid #eeeeee;

  & .ordersTableHeader {
    display: flex;
    width: 100%;
    background: #036077;
    padding: (12px * 1.5) 0;
    color: white;
    text-transform: uppercase;

    & .headerData {
      flex: 1 1 20%;
      text-align: center;
    }
  }

  & .ordersTableRow {
    display: flex;
    width: 100%;
    padding: (12px * 1.5) 0;

    &:nth-of-type(odd) {
      background: #eeeeee;
    }

    & .tableItem {
      flex: 1 1 20%;
      text-align: center;
    }

    & .openOrderLink {
      text-decoration: none;
      color: #036077;

      &:hover {
        color: #098aaa;
      }
    }
  }
}
