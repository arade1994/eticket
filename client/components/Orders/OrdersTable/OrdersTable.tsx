import { useCallback } from "react";
import Link from "next/link";

import { type Order } from "../../../types/order";
import { type User } from "../../../types/user";
import { formatDateToHHMM } from "../../../utils/date";

import classes from "./OrdersTable.module.scss";

const headerLabels = [
  "Ticket title",
  "Price($)",
  "User",
  "Status",
  "Expires at",
  "Actions",
];

interface Props {
  orders: Order[];
  users: User[];
}

const OrdersTable: React.FC<React.PropsWithChildren<Props>> = ({
  orders,
  users,
}) => {
  const getUserFullName = useCallback(
    (userId: string) => {
      const user = users?.find((user) => user.id === userId);

      return `${user?.firstName} ${user?.lastName}`;
    },
    [users]
  );

  return (
    <table className={classes.ordersTable}>
      <thead className={classes.ordersTableHeader}>
        {headerLabels.map((label) => (
          <th key={label} className={classes.headerData}>
            {label}
          </th>
        ))}
      </thead>
      <tbody>
        {orders.map((order) => (
          <tr key={order.id} className={classes.ordersTableRow}>
            <td className={classes.tableItem}>{order?.ticket?.title}</td>
            <td className={classes.tableItem}>{order?.ticket?.price}</td>
            <td className={classes.tableItem}>
              {getUserFullName(order?.userId)}
            </td>
            <td className={classes.tableItem}>{order?.status}</td>
            <td className={classes.tableItem}>
              {formatDateToHHMM(order?.expiresAt)}
            </td>
            <td className={classes.tableItem}>
              <Link
                className={classes.openOrderLink}
                href={`/orders/${order.id}`}
              >
                Open
              </Link>
            </td>
          </tr>
        ))}
      </tbody>
    </table>
  );
};

export default OrdersTable;
