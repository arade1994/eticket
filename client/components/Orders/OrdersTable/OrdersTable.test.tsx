import { render } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import OrdersTable from "./OrdersTable";
import mockDb from "../../../mock/db.json";
import { Order } from "../../../types/order";

const mockOrders = mockDb.orders.splice(0, 2);
const mockUsers = mockDb.users;

describe("<OrdersTable />", () => {
  test("it should render selected orders inside a table", () => {
    const { baseElement } = render(
      <OrdersTable
        orders={mockOrders as unknown as Order[]}
        users={mockUsers}
      />
    );

    expect(baseElement).toMatchSnapshot();
  });
});
