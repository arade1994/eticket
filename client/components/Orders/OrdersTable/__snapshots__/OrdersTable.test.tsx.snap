// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<OrdersTable /> > it should render selected orders inside a table 1`] = `
<body>
  <div>
    <table
      class="_ordersTable_eb8f56"
    >
      <thead
        class="_ordersTableHeader_eb8f56"
      >
        <th
          class="_headerData_eb8f56"
        >
          Ticket title
        </th>
        <th
          class="_headerData_eb8f56"
        >
          Price($)
        </th>
        <th
          class="_headerData_eb8f56"
        >
          User
        </th>
        <th
          class="_headerData_eb8f56"
        >
          Status
        </th>
        <th
          class="_headerData_eb8f56"
        >
          Expires at
        </th>
        <th
          class="_headerData_eb8f56"
        >
          Actions
        </th>
      </thead>
      <tbody>
        <tr
          class="_ordersTableRow_eb8f56"
        >
          <td
            class="_tableItem_eb8f56"
          >
            Football Match
          </td>
          <td
            class="_tableItem_eb8f56"
          >
            230.00
          </td>
          <td
            class="_tableItem_eb8f56"
          >
            Ante Rade
          </td>
          <td
            class="_tableItem_eb8f56"
          >
            created
          </td>
          <td
            class="_tableItem_eb8f56"
          >
            19:37
          </td>
          <td
            class="_tableItem_eb8f56"
          >
            <a
              class="_openOrderLink_eb8f56"
              href="/orders/5ccb"
            >
              Open
            </a>
          </td>
        </tr>
        <tr
          class="_ordersTableRow_eb8f56"
        >
          <td
            class="_tableItem_eb8f56"
          >
            Bethoveen Concerto
          </td>
          <td
            class="_tableItem_eb8f56"
          >
            540.00
          </td>
          <td
            class="_tableItem_eb8f56"
          >
            Ante Rade
          </td>
          <td
            class="_tableItem_eb8f56"
          >
            created
          </td>
          <td
            class="_tableItem_eb8f56"
          >
            19:37
          </td>
          <td
            class="_tableItem_eb8f56"
          >
            <a
              class="_openOrderLink_eb8f56"
              href="/orders/6253"
            >
              Open
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</body>
`;
